src/mir.md

```diff
@@ -1,26 +1,229 @@
 # The MIR (Mid-level IR)
 
-MIR is Rust's _Mid-level Intermediate Representation_. It is constructed from
-HIR (described in an earlier chapter).
+MIR is Rust's _Mid-level Intermediate Representation_. It is
+constructed from HIR (described in an earlier chapter). MIR was
+introduced in [RFC 1211]. It is a radically simplified form of Rust
+that is used for certain flow-sensitive safety checks -- notably the
+borrow checker! -- and also for optimization and code generation.
+
+If you'd like a very high-level introduction to MIR, as well as some
+of the compiler concepts that it relies on (such as control-flow
+graphs and desugaring), you may enjoy the
+[rust-lang blog post that introduced MIR][blog].
+
+[blog]: https://blog.rust-lang.org/2016/04/19/MIR.html
+
+## Introduction to MIR
 
 MIR is defined in the [`src/librustc/mir/`][mir] module, but much of the code
 that manipulates it is found in [`src/librustc_mir`][mirmanip].
 
+[RFC 1211]: http://rust-lang.github.io/rfcs/1211-mir.html
+
+Some of the key characteristics of MIR are:
+
+- It is based on a [control-flow graph][cfg].
+- It does not have nested expressions.
+- All types in MIR are fully explicit.
+
+[cfg]: ./background.html#cfg
+
+## Key MIR vocabulary
+
+This section introduces the key concepts of MIR, summarized here:
+
+- **Basic blocks**: units of the control-flow graph, consisting of:
+  - **statements:** actions with one successor
+  - **terminators:** actions with potentially multiple successors; always at the end of a block
+  - (if you're not familiar with the term basic block, see the [MIR background chapter][bg])
```

Comment:
Perhaps _italicize_ "basic block"?

Reply:
Also, is it now the "background chapter"? And I think you need to define link [bg] somewhere...